lib/src/server/server.dart

```diff
@@ -228,13 +236,23 @@ class Server extends ConnectionServer {
       if (socket.address.type != InternetAddressType.unix) {
         socket.setOption(SocketOption.tcpNoDelay, true);
       }
+
       X509Certificate? clientCertificate;
+
       if (socket is SecureSocket) {
         clientCertificate = socket.peerCertificate;
       }
-      final connection = ServerTransportConnection.viaSocket(socket,
-          settings: http2ServerSettings);
-      serveConnection(connection, clientCertificate);
+
+      final connection = ServerTransportConnection.viaSocket(
+        socket,
+        settings: http2ServerSettings,
+      );
+
+      serveConnection(
+        connection: connection,
+        clientCertificate: clientCertificate,
+        remoteAddress: socket.remoteAddress,
```

Comment:
This line, fetching socket.remoteAddress, has created an uncaught exception in the case where the socket is already closed when we reach this point.

Previously, the server worked if a server connection was requested for a closed socket, and did not throw an exception.  I had a server running for 16 months without failing using the previous code, and running it with this code failed in less than 8 hours.